cli/pcluster/config/validators.py

```diff
@@ -1083,14 +1084,17 @@ def queue_validator(section_key, section_label, pcluster_config):
     compute_resource_labels = str(queue_section.get_param_value("compute_resource_settings") or "").split(",")
 
     def check_queue_xor_cluster(param_key):
-        cluster_section_value = pcluster_config.get_section("cluster").get_param_value(param_key)
-        param_value = queue_section.get_param_value(param_key)
-        if param_value is not None:
-            if cluster_section_value is not None:
+        """Check that the param is not used in both queue and cluster section."""
+        config_parser = pcluster_config.config_parser
+        if config_parser:
+            # This check is performed only if the configuration is loaded from file.
+            queue_param_in_config_file = config_parser.has_option(
```

Comment:
Why are we searching in the config_parser and not in the pcluster_config object itself?

Reply:
As per our conversation, this validation must be performed at config_file level. I added a FIXME note to remind us to improve the validation mechanism by allowing validator to be explicitely attached to a specific validation phase.